src/sql/workbench/contrib/notebook/browser/cellViews/linkHandler.directive.ts

```diff
@@ -61,11 +62,19 @@ export class LinkHandlerDirective {
 			if (uri.fragment && uri.fragment.length > 0 && uri.fsPath === this.workbenchFilePath.fsPath) {
 				this.notebookService.navigateTo(this.notebookUri, uri.fragment);
 			} else {
-				if (this.forceOpenExternal(uri)) {
-					this.openerService.open(uri, { openExternal: true }).catch(onUnexpectedError);
+				let newUri: URI | undefined;
+				let path = dirname(this.notebookUri.path);
+				path = 'file://' + path + '//' + basename(uri.path);
```

Comment:
What happens here if you have a really complex path to another directory? Seems like just the basename would leave out a lot of path information. For example, say I was using a notebook in Documents, but wanted to use a relative link to a notebook in a folder that itself is in the Downloads folder.

Reply:
If it's a relative path, could we not just append it directly, and use path.resolve to convert to an absolute path?